src/Compilers/CSharp/Portable/Symbols/Tuples/TupleMethodSymbol.cs

```diff
@@ -24,10 +24,13 @@ public TupleMethodSymbol(TupleTypeSymbol container, MethodSymbol underlyingMetho
             Debug.Assert(underlyingMethod.ConstructedFrom == (object)underlyingMethod);
             _containingType = container;
 
-            TypeMap.Empty.WithAlphaRename(underlyingMethod, this, out _typeParameters);
-            _underlyingMethod = underlyingMethod.ConstructIfGeneric(TypeArguments);
+            bool nonNullTypes = GetNonNullTypes(underlyingMethod);
```

Comment:
I didn't understand why we use the NonNullTypes context from the underlying method. Could you walk me through this change?

Reply:
`TupleMethodSymbol` is a view on `underlyingMethod` so we should delegate `NonNullTypes` to that method. The reason for passing the value explicitly here is that the value we would use otherwise is `_underlyingMethod` which is not initialized at this point.

---
In reply to: [203884332](https://github.com/dotnet/roslyn/pull/28687#discussion_r203884332) [](ancestors = 203884332)

Reply:
Let's try to Skype tomorrow to walk through this. I'm still having trouble with the `OriginalDefinition.NonNullTypes` business (not just for tuple methods, but more generally for generics).

If you have `T Copy<T>(T t)`, as an original definition, I can see that it has a `NonNullTypes` context where it is defined. This sets the context for interpreting the types of parameters, return and locals.

But then when it is used as `Copy<string>`, I'm not sure that it makes sense to ask what `NonNullTypes` context applies to this substituted method. Who needs that information?